xarray/tests/test_dataarray.py

```diff
@@ -3024,6 +3024,47 @@ def test_to_pandas(self):
         with raises_regex(ValueError, 'cannot convert'):
             DataArray(np.random.randn(1, 2, 3, 4, 5)).to_pandas()
 
+    def test_index_immutability(self):
+        # Test for immutability of Pandas index from #2949
+        index = pd.Index(list('abcdefghij'), name='persistent')
+        series = pd.Series(np.random.randn(10), index)
+        array = xr.DataArray.from_series(series)
+        dataset = xr.Dataset({'myvar': array})
+
+        # Utility function that allows us to consistently many different paths
+        # to accessing the index.
+        def test_index(get_index):
+            # Test `index.data`
+            with pytest.raises(Exception):
+                get_index().data = np.random.randn(10)
+
+            # Test `index.dtype`
+            with pytest.raises(Exception):
+                get_index().dtype = np.float64
```

Comment:
These first two tests (for `index.data` and `index.dtype`) looks like they're testing implementation details of `pandas.Index`, rather than how xarray wraps it. I think we should probably leave this sort of testing to pandas itself.

Reply:
My rationale was: every test uses a `get_index()` function, so every test tests how Xarray wraps Pandas. Given that we're changing how Pandas treats `Index.name`, it's not a given that we're not changing how Pandas treats `Index.dtype`.